### PR TITLE
Rename "context" to "bundle"

### PR DESCRIPTION
**Pull request by stasm**
API changes:

- Rename `MessageContext` to `FluentBundle`.
- Rename `MessageArgument` to `FluentType`.
- Rename `MessageNumberArgument` to `FluentNumber`.
- Rename `MessageDateTimeArgument` to `FluentDateTime`.
- Rename `mapContextSync` to `mapBundleSync`.
- Rename `mapContextSync` to `mapBundleSync`.
- Rename the `messages` prop of `<LocalizationProvider>` to `bundles`.

Other internal changes:

- Rename all uses of `ctx`, `cx`, `mcx`, `mc`, `ctxs` and `contexts` to `bundle` or `bundles`.
- Rename `Localization`'s `somethingFromContext` methods to `somethingFromBundle`.
- Rename `Localization.ctxts` to `bundles`.
- Rename `generateMessages` to `generateBundles` in documentation and examples.

### CODE REVIEW

File: fluent-gecko/README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ natural language.
 The `fluent-gecko` build system produces `*.jsm` files which are ready to be
 used as Gecko modules.
 
-`Fluent.jsm` exports the `MessageContext` constructor which provides the
+`Fluent.jsm` exports the `FluentBundle` constructor which provides the
```
Comment (stasm) [inline fluent-gecko/README.md]: @zbraniecki Do we have a `Fluent.jsm` in `fluent-gecko`?

File: fluent-gecko/makefile [hunk 2]
```diff
@@ -2,9 +2,9 @@ PACKAGE := fluent-gecko
 
 include ../common.mk
 
-build: MessageContext.jsm Localization.jsm DOMLocalization.jsm l10n.js
+build: FluentBundle.jsm Localization.jsm DOMLocalization.jsm l10n.js
 
-MessageContext.jsm: $(SOURCES)
+FluentBundle.jsm: $(SOURCES)
```
Comment (stasm) [inline fluent-gecko/makefile]: @zbraniecki What's your preference wrt. renaming this JSM? Should we do it?

File: fluent-gecko/README.md (hunk 1 above)
Comment (zbraniecki) [reply fluent-gecko/README.md]: No, we have MessageContext.jsm

File: fluent-gecko/makefile (hunk 2 above)
Comment (zbraniecki) [reply fluent-gecko/makefile]: No preference. Happy to follow your preference.

File: fluent-dom/src/localization.js [hunk 3]
```diff
@@ -6,21 +6,21 @@ import { CachedAsyncIterable } from "cached-iterable";
...
+   *                                           generator over FluentBundles
    *
    * @returns {Localization}
    */
   constructor(resourceIds = [], generateMessages) {
     this.resourceIds = resourceIds;
     this.generateMessages = generateMessages;
-    this.ctxs = CachedAsyncIterable.from(
+    this.bundles = CachedAsyncIterable.from(
       this.generateMessages(this.resourceIds));
```
Comment (zbraniecki) [inline fluent-dom/src/localization.js]: generateBundles? :)

File: fluent-gecko/README.md (hunk 1 above)
Comment (zbraniecki) [reply fluent-gecko/README.md]: No, of course not. Why would we? :( Sorry, Yeah, we have MessageContext.jsm, and I'm happy to rename it to FluentBundle.jsm

File: fluent-gecko/makefile (hunk 2 above)
Comment (zbraniecki) [reply fluent-gecko/makefile]: No preference. Open to renaming it. At least to Fluent.jsm :)

File: fluent-sequence/test/fallback_async_test.js [hunk 4]
```diff
@@ -1,121 +1,121 @@
...
-    ctx1 = new MessageContext();
-    ctx1._setMessages(['bar']);
-    ctx2 = new MessageContext();
-    ctx2._setMessages(['foo', 'bar']);
+    bundle1 = new FluentBundle();
+    bundle1._setMessages(['bar']);
+    bundle2 = new FluentBundle();
+    bundle2._setMessages(['foo', 'bar']);
 
     generateMessages = async function *generateMessages() {
```
Comment (zbraniecki) [inline fluent-sequence/test/fallback_async_test.js]: `generateBundles`?

File: fluent-dom/README.md [hunk 5]
```diff
@@ -48,8 +48,8 @@ class that provides just the API needed to format messages in the running code.
 import { Localization } from 'fluent-dom'
 
 function *generateMessages() {
```
Comment (zbraniecki) [inline fluent-dom/README.md]: `generateBundles`?

File: fluent-dom/src/localization.js (hunk 3 above)
Comment (stasm) [reply fluent-dom/src/localization.js]: Ah, yeah, I forgot `fluent-dom` had those too.

**Pull request merged**